### ISSUE TITLE
Should there be a build_policy = "local"?

### ISSUE THREAD

**Issue by bilke** [conan-io/conan#3209]
I would like to have the possibility that a package should be built locally once but in subsequent `conan install`-runs be used from the cache. The `"always"`-policy rebuilds the package again and again.

A local build policy would be helpful when I want to make use of machine-specific build optimizations.

**Comment by memsharded** [comment]
What is wrong with the ``build_policy="missing"``? That builds the first time only, then reuse. 
The ``always`` policy is for cases when you want to force the rebuild, always, as its name says :) Mostly useful if you are developing some repo and want to force always getting the latest source changes, but probably shouldn't be used for typicaly consumption of packages.

**Comment by bilke** [comment]
But when using `build_policy="missing"` then it downloads binary packages from the remote if they exist right?

**Comment by memsharded** [comment]
Yes, it does, if there are binaries. If you don't want them, just don't upload the binaries :)

The problem with the model you are suggesting is that such a policy would make the upstream binaries unusable, because the first time they will be built, not downloaded. So I think that ``missing`` + not uploading the binaries could be what you need, isn't it?

**Comment by PhilipMourdjis** [comment]
In my case, The binaries are provided by a third party so there is no option to not upload them. Building them locally works just fine. 

For reference I am trying to use the mongo-cxx-driver here: https://github.com/bisect-pt/conan-mongo-cxx-driver with c++17 and boost options enabled using gcc 7.3 x86_64 with -D_GLIBCXX_USE_CXX11_ABI=0 set in CMake (imposed by a key third party dependency that can't be changed for the foreseeable future)

My current workaround is calling conan install with -b missing and -b mongo-cxx-driver but this is not great in the long run.

Perhaps it is a problem with identifying binaries as compatible when they are not?

I was unsure about opening another issue on this or not but I think a build policy local as described would fix the issues I am seeing, of course I could just be doing something wrong and I would be happy to be pointed at a better way of fixing this issue.

**Comment by PhilipMourdjis** [comment]
I have an application that is reliant on the mongo-cxx-driver and this dependency is expressed in  conan/conanfile.py file. If I run `conan install conan -if build -b missing -s compiler.libcxx=libstdc++` conan will grab pre-built binaries from bisect (where the mongo-cxx-driver conan package is hosted). These don't work. Forcing a local build of that package with `conan install conan -if build -b missing -b mongo-cxx-driver -s compiler.libcxx=libstdc++` does work. 

> It is possible that that recipe might be missing some setting, (the CXX11_ABI is modeled by settings.compiler.libcxx, maybe the recipe needs to add the cppstd setting?

I have contributed to the mongo-cxx-driver recipe before but I'm not sure how I would go about doing this, it sounds like it could be the solution.

**Comment by memsharded** [comment]
This use case is supported now in 2.0 with:

- ``build_policy = "missing"``
- ``upload_policy = "skip"``

That way packages be built only once, without further command line ``--build`` arguments. And no binaries will be uploaded to the servers.

With that, we can close this ticket, there could be another separate issue with the binary compatibility, but that would require its own ticket. Thanks!
